File: frontend/src/models/sale.ts
import { BikeSale } from './bikeSale';
import { Customer } from './customer';

export interface Sale {
  id: number;
  totalSale: number;
  netSale: number;
  discountPercentage: number;
  soldAt: string;
  updatedAt: string;
  customer: Customer;
  bikes: BikeSale[];
}

File: api/app/models/sale.rb
class Sale < ApplicationRecord
    validates :payment_method, inclusion: { in: ["cash", "credit/debit"],
        message: "%{value} is not a valid payment method" }
    validates :discount_percentage, numericality: { greater_than_or_equal_to: 0, 
        less_than_or_equal_to: 100, message: "is not in valid range" }

    belongs_to :customer
    has_many :bike_sales
    has_many :bikes, through: :bike_sales
end
